File: 1700. Number of Students Unable to Eat Lunch.go
func countStudents(students []int, sandwiches []int) int {
	count := 0
	for len(students) != 0 && count < len(students) {
		if students[0] == sandwiches[0] {
			students = students[1:]
			sandwiches = sandwiches[1:]
			count = 0
		} else {
			students = append(students, students[0])
			students = students[1:]
			count += 1
		}
	}
	return len(students)
}

File: 1700. Number of Students Unable to Eat Lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        Deque<Integer> sandwichQueue = new LinkedList<>();
        Deque<Integer> studentQueue = new LinkedList<>();
        for (int i = 0; i < students.length; i++) {
            studentQueue.add(students[i]);
            sandwichQueue.add(sandwiches[i]);
        }
        while (!studentQueue.isEmpty() && count < studentQueue.size()) {
            if (studentQueue.peek() == sandwichQueue.peek()) {
                studentQueue.poll();
                sandwichQueue.poll();
                count = 0;
            } else {
                studentQueue.add(studentQueue.poll());
                count++;
            }
        }
        return studentQueue.size();
    }
}